### ISSUE TITLE
JFXCheckBox indeterminate mark issue

### ISSUE THREAD

**Issue by mordechaim** [jfoenixadmin/JFoenix#517]
https://github.com/jfoenixadmin/JFoenix/blob/e1d39638d3c71c85b1bdd11ffe6cdd7fa1e4120d/jfoenix/src/main/java/com/jfoenix/skins/JFXCheckBoxSkin.java#L67

In the recent addition of indeterminate mark, it is not possible to set mark fill by `setColorChecked()` method. After checking the JavaFX CSS reference, it is possible to style it by:

    .check-box:indeterminate .box .mark {
    }

This didn't work in `JFXCheckBox`. After hunting down the source, I found that this project outlies the default style class and uses its own `indeterminate-mark` style class.

Unless this is well documented, this makes it very hard to follow.

A simple quick-fix would be to say:

     indeterminateMark.getStyleClass().setAll("mark", "indeterminate-mark"); 

Also the `setColorChecked()` should apply to this too.

**Comment by mordechaim** [comment]
Also noticed that checked color would still leave the default box color, which i have to change with css by `-fx-border-color`.

**Comment by jfoenixadmin** [comment]
Thanks for pointing out the issue 👍 , will update it to keep consistency with javafx styling.

**Comment by jfoenixadmin** [comment]
Hello,
Well "indeterminate-mark" has different shape and animation than the selection mark. thus it's better to keep them separate. 
Regarding the box color, it is linked to unchecked color property.

Regards,

**Comment by mordechaim** [comment]
I'm not at my computer now, but i remember setting both checked and
unchecked colors, but the box would still have the original (darkcyan)
color "when checked".

How weird it sounds, this won't happen if it's unchecked.

**Comment by jfoenixadmin** [comment]
can you provide a sample example to demonstrate the issue?

**Comment by mordechaim** [comment]
```
import com.jfoenix.controls.JFXCheckBox;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.layout.BorderPane;
import javafx.scene.paint.Color;
import javafx.stage.Stage;

public class JFXCheckBoxTest extends Application {

	@Override
	public void start(Stage primaryStage) {
		
		JFXCheckBox check = new JFXCheckBox("Check me!");
		System.out.println("Checked Color: " + check.getCheckedColor());
		System.out.println("Unchecked Color: " + check.getUnCheckedColor());
		
		check.setCheckedColor(Color.RED);
		check.setUnCheckedColor(Color.RED);
		
		System.out.println("Checked Color: " + check.getCheckedColor());
		System.out.println("Unchecked Color: " + check.getUnCheckedColor());
		
		final BorderPane root = new BorderPane();
		root.setCenter(check);
		primaryStage.setScene(new Scene(root, 600, 400));
		primaryStage.show();
	}


	public static void main(String[] args) {
		launch(args);
	}
}

```

Output:
```

Checked Color: 0x0f9d58ff
Unchecked Color: 0x5a5a5aff
Checked Color: 0xff0000ff
Unchecked Color: 0xff0000ff

```

Unchecked
![capture](https://user-images.githubusercontent.com/12547785/33344913-85e80a44-d458-11e7-8e57-d36974d7c5e4.PNG)

Checked

![capture](https://user-images.githubusercontent.com/12547785/33344935-93cfa07c-d458-11e7-9e6d-a10a155d79e4.PNG)


As you can see this green box border doesn't follow the coloring. only `-fx-border-color` will fix it.

**Comment by mordechaim** [comment]
You can also see that unchecked color has no effect at all...

**Comment by jfoenixadmin** [comment]
Hello,
it seems this issue is related to JavaFX css engine. User agent stylesheet is not applied correctly when refereeing a custom styleable property. 
I updated the skin to change colors in code instead of depending on the user agent stylesheet.

Regards,
